File: java/lib/src/test/java/org/sammancoaching/library/MemberTest.java
package org.sammancoaching.library;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.function.Executable;

import static org.junit.jupiter.api.Assertions.assertThrows;

class MemberTest {

    @Test
    void testMemberNoId() {
        Executable createMemberWithoutId = () -> new Member(" ", "Alice");

        assertThrows(IllegalArgumentException.class, createMemberWithoutId);
    }

    @Test
    void testMemberNoName() {
        Executable createMemberWithoutName = () -> new Member("MEMBER-001", " ");

        assertThrows(IllegalArgumentException.class, createMemberWithoutName);
    }
}

File: csharp/Library.Tests/MemberTest.cs
using NUnit.Framework;
using SammanCoaching.Library;

namespace Library.Tests;

public class MemberTest
{
    [Test]
    public void testMemberNoId()
    {
        var memberId = " ";
        var name = "Alice";

        Action act = () => new Member(memberId, name);

        Assert.Throws<ArgumentException>(act);
    }

    [Test]
    public void testMemberNoName()
    {
        var memberId = "MEMBER-001";
        var name = " ";

        Action act = () => new Member(memberId, name);

        Assert.Throws<ArgumentException>(act);
    }
}
